File: 015.c
#include <stdio.h>
int paths(long int x, long int y){
    if (x == 1 || y == 1) return x + y;
    return paths(x - 1, y) + paths(x, y - 1);
};
int main(){
    printf("paths(20, 20): %d\n", paths(20, 20));
    return 0;
};

File: 015.js
require('./common/euler').run(function () {
    var known = {},
        paths = function (x, y) {
            if (x === 1 || y === 1) return x + y;
            var key = x + ',' + y, alt_key = y + ',' + x;
            if (key in known) return known[key];
            return known[key] = known[alt_key] = paths(x - 1, y) + paths(x, y - 1);
        };
    return paths(20, 20);
});